LEDControl_v2/LEDControl_v2/APP/app.c

```diff
@@ -0,0 +1,154 @@
+/**
+ *  FILE DESCRIPTION
+ *  -------------------------------------------------------------------------------------------------------------------
+ *      @File  			app.h
+ *      @date			18/6/2023	2:26 PM
+ *  	@brief			interface file for application
+ *		@author			Ahmad Hesham
+ *      @email			ahmedhesham606@icloud.com
+ */
+
+
+
+#include "app.h"
+#include "led_interface.h"
+#include "button_interface.h"
+#include "systick_handler_interface.h"
+/**----------------------------------------------------------
+ *  STATIC FUNCTION PROTOTYPES
+ *----------------------------------------------------------*/
+static void app_button_task(void);
+static void app_systick_task(void);
+static void app_led_task(void);
+typedef enum
+{
+	FIRST_PRESS,
+	SECOND_PRESS,
+	THIRD_PRESS,
+	FOURTH_PRESS,
+	FIFTH_PRESS
+}enu_app_pressing_state_t_;
+
+/**----------------------------------------------------------
+ *  GLOBAL FUNCTION IMPLEMENTAIONS
+ *----------------------------------------------------------*/
+/**
+ *	@brief				This function is used to initialize drivers used
+ *	@param [in]		none
+ *	@param [out]	none
+ *	@return				none
+ */
+void app_init(void)
+{
+	led_init();
+	button_init(NULL_PTR);
+	handler_init_systick(APP_SYSTICK_TIME);
+	handler_set_callback_systick(app_systick_task);
+}
+
+/**----------------------------------------------------------
+ *  STATIC FUNCTION IMPLEMENTATION
+ *----------------------------------------------------------*/
+/**
+ *	@brief				This function is the called to perform LED task
+ *	@param [in]		none
+ *	@param [out]	none
+ *	@return				none
+ */
+static void app_led_task(void)
+{
+	static enu_app_pressing_state_t_ enu_pressing_state = FIRST_PRESS;
+	switch (enu_pressing_state)

```

Comment:
Switch case should have only one break

Reply:
I think there are two solutions:
1. The easy way to replace the switch case with if condition.
2. The hard way is to use Fall-through switch cases.